lib/brains/webhooks.py

```diff
@@ -30,6 +33,27 @@ def process(self):
 
         log.info('Processing event: {0}'.format(self.webhook.kind))
         method()
+        self.processed = True
+
+    @property
+    def data(self):
+        if not self.processed:
+            return
+
+        subscription = self.get_subscription()
+        return Namespaced(
+            LocalReceipt(
+                subscription=subscription,
+                paymethod=subscription.paymethod,
+                # TOOD: this does not feel very thread safe.
+                # Transactions on a subscription are:
+                #   Transactions associated with the subscription,
+                #   sorted by creation date with the most recent first.
+                transaction=subscription.get_transactions(
+                    ids=self.transaction_ids).first()
```

Comment:
This should also be filtering by the transaction type, eg: completed transactions. I'm not sure why braintree would send the most recent transaction as a failed one but I don't know.

Also feel like I should do something smart if there's not transaction at all.